test/hierarchical/test_ordinal.py

```diff
@@ -0,0 +1,310 @@
+                1.99999999e00,
+                2.58823529e00,
+                4.00689557e00,
+                4.59542030e00,
+                6.23751538e00,
+                6.82578436e00,
+                8.40233862e00,
+                8.99060060e00,
+                1.10767572e01,
+            ]
+        ),
+        REDUCED: np.asarray(
+            [1.99999999, 4.00689557, 6.23751538, 8.40233862, 11.0767572]
+        ),
+    }
+
+    simulation = timepoints
+    data = np.full(simulation.shape, np.nan)
+
+    par_types = ['cat_lb', 'cat_ub']
+
+    categories = [1, 2, 3, 4, 5, 1, 2, 3, 4, 5]
+    inner_parameter_ids = [
+        f'{par_type}_{category}'
+        for par_type in par_types
+        for category in set(categories)
+    ]
+
+    # Get data masks for inner parameters
+    masks = []
+    for category in categories:
+        new_mask = np.full(data.shape, False)
+        new_mask[2 * (category - 1) : 2 * category] = True
+        masks.append([new_mask])
+
+    # Construct inner parameters
+    inner_parameters = [
+        OptimalScalingParameter(
+            inner_parameter_id=inner_parameter_id,
+            inner_parameter_type=InnerParameterType.OPTIMAL_SCALING,
+            scale=LIN,
+            lb=-np.inf,
+            ub=np.inf,
+            ixs=mask,
+            group=1,
+            category=inner_parameter_category,
+        )
+        for inner_parameter_id, mask, inner_parameter_category in zip(
+            inner_parameter_ids, masks, categories
+        )
+    ]
+
+    # Construct inner problem
+    inner_problem = OptimalScalingProblem(
+        xs=inner_parameters, data=[data], method=STANDARD
+    )
+
+    return inner_problem, expected_inner_parameter_values, simulation
+
+
+def test_optimal_scaling_solver():
+    """Test the Optimal scaling solver."""
+    (
+        inner_problem,
+        expected_values,
+        simulation,
+    ) = _inner_problem_exp()
+
+    rtol = 1e-3
+
+    solver = OptimalScalingInnerSolver(
+        options={'method': STANDARD, 'reparameterized': False}
+    )
+
+    standard_result = solver.solve(
+        problem=inner_problem,
+        sim=[simulation],
+    )[0]
+
+    assert np.all(
+        np.isclose(standard_result['x'], expected_values[STANDARD], rtol=rtol)
+    )
+    assert np.all(np.isclose(standard_result['fun'], 0, rtol=rtol))
+    assert np.all(np.isclose(standard_result['jac'], 0, rtol=rtol))
+
+    solver = OptimalScalingInnerSolver(
+        options={'method': REDUCED, 'reparameterized': False}
+    )
+
+    reduced_result = solver.solve(
+        problem=inner_problem,
+        sim=[simulation],
+    )[0]
+
+    assert np.all(
+        np.isclose(reduced_result['x'], expected_values[REDUCED], rtol=rtol)
+    )
+    assert np.all(np.isclose(reduced_result['fun'], 0, rtol=rtol))
+    assert np.all(np.isclose(reduced_result['jac'], 0, rtol=rtol))
+
+
+def test_surrogate_data_analytical_calculation():
+    """Test analytical calculation of surrogate data.
```

Comment:
:+1:

Reply:
would be even better with formulas that reproduce how the analytical values were calculated, but ok

Reply:
`get_surrogate_all` and `compute_interval_constraints` are the functions of analytical formulas for everything that's being computed analytically in the inner_solver. So those are the ones I'm testing. 
Do you mean to get the `expected_values` with new, rewritten formulas in `test_ordinal.py`?

Reply:
yes, explicit formulas for expected values instead of "magic" numbers would improve credibility, but only if easily possible. otherwise we trust that you calculated these correctly :D

Reply:
Yes, it's easy to do. Fixed in next commit.